aopy/analysis/base.py

```diff
@@ -2561,6 +2561,180 @@ def simulate_ideal_trajectories(targets, origin=[0.0, 0.0, 0.0], resolution=1000
+
+        ax2.set_title(f'Baseline Speed')
+        ax2.set_xlabel(f'time (s)')
+        ax2.set_ylabel(f'speed (cm/s)')
+        for initial,submov in zip(initial_speed[:8], submovement_speed[:8]):
+            initial_time = np.arange(len(initial)) / 1000
+            submovement_time = np.arange(len(initial),len(initial)+len(submov)) / 1000
+            ax2.plot(initial_time, initial, color='b', alpha=0.25)
+            ax2.plot(submovement_time, submov, color='r', alpha=0.25)
+            
+    .. image:: _images/reach_submovements.png
+        
+    """
+    
+    if origins is None:
+        origins = np.zeros((len(trajectories), 3), dtype=float)
+                   
+    if len(trajectories) != len(origins):
+        raise ValueError("Number of trials must be consistent between trajectories and origins.")
+    
+    from scipy.signal import find_peaks, peak_prominences
+    
+    split_indices = []
+    
+    for T,traj in enumerate(trajectories):
+    
+        trajectory = np.asarray(traj)
+        speed = np.asarray(speeds[T])
+        if trajectory.shape[0] != speed.shape[0]:
+            raise ValueError("Position and speed vectors must be the same length.")
+
+        N = len(speed)
+        if N < 3:
+            split_indices.append(len(traj))  # not enough samples to define peaks & troughs
+            continue
+
+        # 1) Find peaks in speed and get their heights
+        peak_idx, peak_properties = find_peaks(speed, height=speed_threshold)
+        if peak_idx.size == 0:
+            split_indices.append(len(traj))
+            continue
+
+        # 2) Compute prominences for those peaks
+        prominences, left_bases, right_bases = peak_prominences(speed, peak_idx)
+
+        # 3) Keep peaks with prominence >= (prominence_fraction * peak_height)
+        peak_heights = peak_properties["peak_heights"]
+        keep = ( prominences >= (prominence_fraction * peak_heights) )
+        peak_idx = peak_idx[keep]
+        if peak_idx.size == 0:
+            split_indices.append(len(traj))
+            continue
+
+        # 4) Find troughs (local minima) in speed
+        trough_idx = find_peaks(-speed)[0]  # peaks of -speed are minima of speed
+        if trough_idx.size == 0:
+            split_indices.append(len(traj))
+            continue
+
+        # 5) For each qualifying peak in chronological order, find the first trough after it.
+        peak_idx = np.sort(peak_idx)
+        no_initial_peak = True
+        for p in peak_idx:
+            # first trough strictly after the peak
+            after = trough_idx[trough_idx > p]
+            if after.size == 0:
+                continue
+            t = after[0]
+
+            # Check the "initial peak" criterion: the trough position is >= distance_threshold
+            if np.linalg.norm(trajectory[t]) >= distance_threshold:
+                split_indices.append(int(t))
+                no_initial_peak = False
+                break
```

Comment:
i'm not sure i understand this part. what are you trying to do here? maybe this can be accomplished with `distance` parameter in find_peaks?